source/code/source/playfab/PlayFabCurlHttpPlugin.cpp

```diff
@@ -110,6 +117,37 @@ namespace PlayFab
         } // UNLOCK httpRequestMutex
     }
 
+    constexpr char requestIdHeaderKey[] = "X-RequestId:";
+    constexpr size_t requestIdheaderKeyLen = _countof(requestIdHeaderKey) - 1;
+    constexpr char whitespace[] = "\t\n\v\f\r ";
+    size_t HeaderCallback(char* buffer, size_t size, size_t nitems, void* userdata)
+    {
+        CallRequestContainer& reqContainer = *static_cast<CallRequestContainer*>(userdata);
+        if (!reqContainer.errorWrapper.RequestId.empty())
+        {
+            // For now, we only care about the RequestId header
+            return nitems * size; // The return expected by curl for this callback
+        }
+
+        // If this header-line is long enough, and the header starts with the key we expect
+        if ((nitems > requestIdheaderKeyLen) && (strncasecmp(buffer, requestIdHeaderKey, requestIdheaderKeyLen) == 0))
+        {
+            // The value is the requestId
+            std::string requestId = std::string(buffer + requestIdheaderKeyLen, nitems - requestIdheaderKeyLen);
+            auto offset = requestId.find_first_not_of(whitespace);
+            if (offset != std::string::npos)
+            {
+                // Trim any whitespace
+                requestId.erase(0, offset);
+                requestId.erase(requestId.find_last_not_of(whitespace) + 1);
```

Comment:
minor: if you reverse the order of these erase calls, then the erase at the beginning might have fewer bytes to copy when it shifts the contents of the string for the erase.